js/auth.js

```diff
@@ -0,0 +1,139 @@
+import { API_BASE_URL } from '../js/constants.js';
+/**
+ * Registers a new user.
+ * @async
+ * @function
+ * @param {string} name - The name of the user.
+ * @param {string} email - The email of the user.
+ * @param {string} password - The password of the user.
+ * @throws Will throw an error if registration fails.
+ * @returns {Promise<void>} No return value.
+ */
+export async function registerUser(name, email, password) {
+    try {
+        const response = await fetch(`${API_BASE_URL}/social/auth/register`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+                name,
+                email,
+                password,
+            }),
+        });
+
+        const errorMessageDiv = document.getElementById('error-message');
+
+        if (!response.ok) {
+            const errorData = await response.json();
+            if (errorData.errors && errorData.errors[0].message === 'Profile already exists') {
+                errorMessageDiv.textContent = 'A profile with this email or username already exists. Please try logging in or use different credentials.';
+            } else {
+                errorMessageDiv.textContent = 'Registration failed. Please try again.';
+                throw new Error('Registration failed');
+            }
+        } else {
+            const data = await response.json();
+            console.log('User registered successfully:', data);
+            errorMessageDiv.textContent = 'User registered successfully!'; // Display a success message
+            errorMessageDiv.classList.remove('text-danger');
+            errorMessageDiv.classList.add('text-success');
+            window.location.href = '/profile/index.html';
```

Comment:
Do you want to log in the user when he or she registers successfully? If yes, then how do you generate the token? According to the API doc., the token is the authentic means of identifying an auth user; without it, the user is not authenticated.

What I am saying essentially is that you should not take the user to the profile page after a successful registration.

Reply:
Very true, thanks coach, i modified it like this:
- Separated Registration and Authentication.
- Handle Token: to automatically log the user in after registration, once the login is successful, the server should return an authentication token. Token is stored and used for subsequent authenticated requests.
- Redirect to Profile: Only after obtaining the token, redirecting the user to the profile page.